File: Python/underautomation/universal_robots/primary_interface/configuration_data_package_event_args.py
import typing
from underautomation.universal_robots.common.controller_box_types import ControllerBoxTypes
from underautomation.universal_robots.common.robot_models import RobotModels
from underautomation.universal_robots.common.robot_sub_types import RobotSubTypes
from underautomation.universal_robots.primary_interface.joint_configuration import JointConfiguration
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import ConfigurationDataPackageEventArgs as configuration_data_package_event_args

class ConfigurationDataPackageEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = configuration_data_package_event_args()
		else:
			self._instance = _internal
	@property
	def v_joint_default(self) -> float:
		return self._instance.VJointDefault
	@v_joint_default.setter
	def v_joint_default(self, value: float):
		self._instance.VJointDefault = value
	@property
	def a_joint_default(self) -> float:
		return self._instance.AJointDefault
	@a_joint_default.setter
	def a_joint_default(self, value: float):
		self._instance.AJointDefault = value
	@property
	def v_tool_default(self) -> float:
		return self._instance.VToolDefault
	@v_tool_default.setter
	def v_tool_default(self, value: float):
		self._instance.VToolDefault = value
	@property
	def a_tool_default(self) -> float:
		return self._instance.AToolDefault
	@a_tool_default.setter
	def a_tool_default(self, value: float):
		self._instance.AToolDefault = value
	@property
	def eq_radius(self) -> float:
		return self._instance.EqRadius
	@eq_radius.setter
	def eq_radius(self, value: float):
		self._instance.EqRadius = value
	@property
	def masterboard_version(self) -> int:
		return self._instance.MasterboardVersion
	@masterboard_version.setter
	def masterboard_version(self, value: int):
		self._instance.MasterboardVersion = value
	@property
	def controller_box_type(self) -> ControllerBoxTypes:
		return ControllerBoxTypes(self._instance.ControllerBoxType)
	@controller_box_type.setter
	def controller_box_type(self, value: ControllerBoxTypes):
		self._instance.ControllerBoxType = value
	@property
	def robot_type(self) -> RobotModels:
		return RobotModels(self._instance.RobotType)
	@robot_type.setter
	def robot_type(self, value: RobotModels):
		self._instance.RobotType = value
	@property
	def robot_sub_type(self) -> RobotSubTypes:
		return RobotSubTypes(self._instance.RobotSubType)
	@robot_sub_type.setter
	def robot_sub_type(self, value: RobotSubTypes):
		self._instance.RobotSubType = value
	@property
	def base(self) -> JointConfiguration:
		return JointConfiguration(self._instance.Base)
	@base.setter
	def base(self, value: JointConfiguration):
		self._instance.Base = value
	@property
	def shoulder(self) -> JointConfiguration:
		return JointConfiguration(self._instance.Shoulder)
	@shoulder.setter
	def shoulder(self, value: JointConfiguration):
		self._instance.Shoulder = value
	@property
	def elbow(self) -> JointConfiguration:
		return JointConfiguration(self._instance.Elbow)
	@elbow.setter
	def elbow(self, value: JointConfiguration):
		self._instance.Elbow = value
	@property
	def wrist1(self) -> JointConfiguration:
		return JointConfiguration(self._instance.Wrist1)
	@wrist1.setter
	def wrist1(self, value: JointConfiguration):
		self._instance.Wrist1 = value
	@property
	def wrist2(self) -> JointConfiguration:
		return JointConfiguration(self._instance.Wrist2)
	@wrist2.setter
	def wrist2(self, value: JointConfiguration):
		self._instance.Wrist2 = value
	@property
	def wrist3(self) -> JointConfiguration:
		return JointConfiguration(self._instance.Wrist3)
	@wrist3.setter
	def wrist3(self, value: JointConfiguration):
		self._instance.Wrist3 = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/PrimaryInterface/ConfigurationDataPackageEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Joint configuration
	/// </summary>
	public class ConfigurationDataPackageEventArgs : PackageEventArgs {

		/// <summary>
		/// Default joint angular speed in rad/s
		/// </summary>
		[PackageDescription("Default joint angular speed in rad/s", PackageUnit.RadianPerSecond)]
public double VJointDefault { get; set; }

		/// <summary>
		/// Default joint acceleration speed in rad/s²
		/// </summary>
		[PackageDescription("Default joint acceleration speed in rad/s²", PackageUnit.RadianPerSecondSquared)]
public double AJointDefault { get; set; }

		/// <summary>
		/// Default TCP speed speed in m/s
		/// </summary>
		[PackageDescription("Default TCP speed speed in m/s", PackageUnit.MeterPerSecond)]
public double VToolDefault { get; set; }

		/// <summary>
		/// Default TCP acceleration speed in m/s²
		/// </summary>
		[PackageDescription("Default TCP acceleration speed in m/s²", PackageUnit.MeterPersSecondSquared)]
public double AToolDefault { get; set; }

		/// <summary>
		/// Equipment radius in meter
		/// </summary>
		[PackageDescription("Equipment radius in meter", PackageUnit.Meter)]
public double EqRadius { get; set; }

		/// <summary>
		/// Masterboard version
		/// </summary>
		[PackageDescription("Masterboard version")]
public int MasterboardVersion { get; set; }

		/// <summary>
		/// Controller box type
		/// </summary>
		[PackageDescription("Controller box type")]
public ControllerBoxTypes ControllerBoxType { get; set; }

		/// <summary>
		/// Model of the robot (UR3, UR5, UR10, UR16)
		/// </summary>
		[PackageDescription("Model of the robot")]
public RobotModels RobotType { get; set; }

		/// <summary>
		/// Robot serie
		/// </summary>
		[PackageDescription("Robot serie")]
public RobotSubTypes RobotSubType { get; set; }

		/// <summary>
		/// Base joint configuration
		/// </summary>
		[PackageDescription("Base joint configuration")]
public JointConfiguration Base { get; set; }

		/// <summary>
		/// Shoulder joint configuration
		/// </summary>
		[PackageDescription("Shoulder joint configuration")]
public JointConfiguration Shoulder { get; set; }

		/// <summary>
		/// Elbow joint configuration
		/// </summary>
		[PackageDescription("Elbow joint configuration")]
public JointConfiguration Elbow { get; set; }

		/// <summary>
		/// Wrist1 joint configuration
		/// </summary>
		[PackageDescription("Wrist1 joint configuration")]
public JointConfiguration Wrist1 { get; set; }

		/// <summary>
		/// Wrist2 joint configuration
		/// </summary>
		[PackageDescription("Wrist2 joint configuration")]
public JointConfiguration Wrist2 { get; set; }

		/// <summary>
		/// Wrist3 (Tool) joint configuration
		/// </summary>
		[PackageDescription("Wrist3 (Tool) joint configuration")]
public JointConfiguration Wrist3 { get; set; }
	}
}
